name_base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class NameDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you use `getter` and `setter` methods on the `@nameable` instance variable as it promotes better code organization, maintainability, and flexibility by providing a standardized way to interact with the state of objects while maintaining control over their access and modification. i.e. `attr_accessor :nameable`